pydantic/types.py

```diff
@@ -530,11 +530,13 @@ def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
-        yield str_validator
         yield cls.validate
 
     @classmethod
-    def validate(cls, value: str) -> 'SecretStr':
+    def validate(cls, value: Any) -> 'SecretStr':
+        if value.__class__ == cls:
```

Comment:
why not isinstance?

Reply:
in case someone uses some subclass in future.

Reply:
I was sure I saw this used somewhere, but I can't find it now, I'll change it to use isinstance instead.

Reply:
This is fixed now.